bignumber.js

```diff
@@ -2369,14 +2369,46 @@
          *
          * 'pow() exponent not an integer: {n}'
          * 'pow() exponent out of range: {n}'
+         *
+         * m {number} Integer|BigNumber
          */
-        P.toPower = P.pow = function (n) {
+        P.toPower = P.pow = function (n, m) {
             var k, y,
                 i = mathfloor( n < 0 ? -n : +n ),
-                x = this;
+                x = this,
+                nIsValidInt = isValidInt( n, -MAX_SAFE_INTEGER, MAX_SAFE_INTEGER, 23, 'exponent' ),
+                hasMod = typeof m !== 'undefined';
+
+            //Implement fast modular exponentiation via
+            //RTL binary method when n >= 0 && m > 0
+            //Returns a BigNumber which equals (x ^ n) % m
+            function powMod(x, n, m, k) {
```

Comment:
It seem unnecessary to define `powMod` as an inner function when it is only called once. It means that the function may be recreated on each call, and adds extra code needlessly.